### PR TITLE
Make ErrorFactories into an object in preparation for its removal. [DPP-606]

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiActiveContractsService.scala [hunk 1]
```diff
@@ -75,12 +71,11 @@ private[apiserver] object ApiActiveContractsService {
...
       loggingContext: LoggingContext,
   ): ActiveContractsService with GrpcApiService = {
-    val errorFactories = ErrorFactories()
-    val field = FieldValidations(ErrorFactories())
+    val field = FieldValidations()
     val service = new ApiActiveContractsService(
       backend = backend,
       metrics = metrics,
-      transactionFilterValidator = new TransactionFilterValidator(errorFactories),
+      transactionFilterValidator = new TransactionFilterValidator(),
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiActiveContractsService.scala:78]: This one as well can be made into an object

**Review by tudor-da — APPROVED** [review]
Nice! I would turn the other 0-args factories into objects as well

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala [hunk 2]
```diff
@@ -37,12 +37,11 @@ import scala.annotation.nowarn
 
 final class CommandsValidator(ledgerId: LedgerId) {
 
-  private val errorFactories = ErrorFactories()
-  private val fieldValidations = FieldValidations(errorFactories)
-  private val valueValidator = new ValueValidator(errorFactories, fieldValidations)
-  private val deduplicationValidator = new DeduplicationPeriodValidator(errorFactories)
+  private val fieldValidations = FieldValidations()
```
[the remark this answers is not in the document]
Comment (pbatko-da) [reply ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala]: Ok, pushed in a new commit

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiActiveContractsService.scala (hunk 1 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiActiveContractsService.scala]: Done

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/deduplication/DeduplicationPeriodSupportSpec.scala [hunk 3]
```diff
@@ -27,39 +27,30 @@ class DeduplicationPeriodSupportSpec
...
     with TestLoggers
-    with ArgumentMatchersSugar {
+    with ArgumentMatchersSugar
+    with ErrorsAssertions {
 
   "using deduplication duration" should {
     "validate and return success" in {
       val fixture = getFixture
       import fixture._
-      when(
```
Comment (pbatko-da) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/deduplication/DeduplicationPeriodSupportSpec.scala:36]: fyi @nicu-da

**Review by nicu-da — APPROVED** [review]

**Pull request merged**